datastore/longhorn.go

```diff
@@ -3171,15 +3174,13 @@ func (s *DataStore) GetInstanceManagerByInstanceRO(obj interface{}) (*longhorn.I
 		nodeID string
 	)
 
-	switch obj.(type) {
+	switch obj := obj.(type) {
 	case *longhorn.Engine:
-		engine := obj.(*longhorn.Engine)
-		name = engine.Name
-		nodeID = engine.Spec.NodeID
+		name = obj.Name
+		nodeID = obj.Spec.NodeID
 	case *longhorn.Replica:
-		replica := obj.(*longhorn.Replica)
-		name = replica.Name
-		nodeID = replica.Spec.NodeID
+		name = obj.Name
+		nodeID = obj.Spec.NodeID
```

Comment:
I am not sure if it's a good idea to redeclare `obj`, perhaps consider using a different name.

Reply:
Good spot. Fixed it.